nova/virt/vmwareapi/vmops.py

```diff
@@ -2861,6 +2861,16 @@ def _do_migrate_disk(self, context, vm_ref, instance, dest_data, flavor):
         rel_spec = target.get_relocate_spec(context, instance, flavor,
                                             factory=factory)
 
+        # Scale the cloned vm down to minimal resources,
+        # so that it fits the source hypervisor as well as the destination.
+        # The actual size will configured after the migration, and should
+        # fit thanks to the scheduler logic
+        client_factory = self._session.vim.client.factory
+        config_spec = client_factory.create('ns0:VirtualMachineConfigSpec')
+        config_spec.memoryMB = 4
```

Comment:
Off-topic, but do you know why a shadow-VM has more than 4 MiB memory?

Reply:
No, I have no idea.